### PR TITLE
fixed nested PreferenceScreens when using Holo_Theme_Dialog

### PR DESCRIPTION
**Pull request by metinkale38**

### CODE REVIEW

File: addons/preferences/src/org/holoeverywhere/preference/PreferenceScreen.java [hunk 1]
```diff
@@ -256,7 +256,8 @@ private void showDialog(Bundle state) {
         if (titleEmpty) {
             dialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
         } else {
-            if (VERSION.SDK_INT >= 11) {
+            if (VERSION.SDK_INT >= 11
+					&& contextTheme != R.style.Holo_Theme_Dialog) {
```
Comment (Prototik) [inline addons/preferences/src/org/holoeverywhere/preference/PreferenceScreen.java:260]: Hm... i don't think that hard reference on same theme - good idea.
You should check a windowActionBar attribute from theme.
Like this:

``` java
TypedArray a = context.obtainStyledAttributes(new int[] { android.R.attr.windowActionBar, R.attr.windowActionBar });
boolean windowActionBar = a.getBoolean(0) || a.getBoolean(1);
a.recycle();
if (VERSION.SDK_INT >= 11 && windowActionBar) {
...
```

**Comment by metinkale38** [comment]
good you havent merged it :) it is not a good practise: causes crashes on <4.0

**Comment by Prototik** [comment]
I suggest that you improve the code, if you have updated your branch to Fix - I have accepted it.

**Comment by metinkale38** [comment]
no the problem is that is causes other problems on e.g. GB, i just tried it on JB before.

on GB i have this crash:
`08-13 20:29:16.821: E/AndroidRuntime(588): FATAL EXCEPTION: main
08-13 20:29:16.821: E/AndroidRuntime(588): java.lang.IllegalStateException: ActionBarView can only be used with android:layout_width="match_parent" (or fill_parent)
08-13 20:29:16.821: E/AndroidRuntime(588):  at com.actionbarsherlock.internal.widget.ActionBarView.onMeasure(ActionBarView.java:813)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.view.View.measure(View.java:8313)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:3138)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.widget.FrameLayout.onMeasure(FrameLayout.java:250)
08-13 20:29:16.821: E/AndroidRuntime(588):  at com.actionbarsherlock.internal.widget.ActionBarContainer.onMeasure(ActionBarContainer.java:182)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.view.View.measure(View.java:8313)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:3138)
08-13 20:29:16.821: E/AndroidRuntime(588):  at org.holoeverywhere.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:594)
08-13 20:29:16.821: E/AndroidRuntime(588):  at org.holoeverywhere.widget.LinearLayout.measureVertical(LinearLayout.java:913)
08-13 20:29:16.821: E/AndroidRuntime(588):  at org.holoeverywhere.widget.LinearLayout.onMeasure(LinearLayout.java:1105)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.view.View.measure(View.java:8313)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:3138)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.widget.FrameLayout.onMeasure(FrameLayout.java:250)
08-13 20:29:16.821: E/AndroidRuntime(588):  at org.holoeverywhere.internal.WindowDecorView.onMeasure(WindowDecorView.java:114)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.view.View.measure(View.java:8313)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:3138)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.widget.FrameLayout.onMeasure(FrameLayout.java:250)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.view.View.measure(View.java:8313)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:3138)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.widget.FrameLayout.onMeasure(FrameLayout.java:250)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.view.View.measure(View.java:8313)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.view.ViewRoot.performTraversals(ViewRoot.java:839)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.view.ViewRoot.handleMessage(ViewRoot.java:1859)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.os.Handler.dispatchMessage(Handler.java:99)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.os.Looper.loop(Looper.java:123)
08-13 20:29:16.821: E/AndroidRuntime(588):  at android.app.ActivityThread.main(ActivityThread.java:3683)
08-13 20:29:16.821: E/AndroidRuntime(588):  at java.lang.reflect.Method.invokeNative(Native Method)
08-13 20:29:16.821: E/AndroidRuntime(588):  at java.lang.reflect.Method.invoke(Method.java:507)
08-13 20:29:16.821: E/AndroidRuntime(588):  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:839)
08-13 20:29:16.821: E/AndroidRuntime(588):  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:597)
08-13 20:29:16.821: E/AndroidRuntime(588):  at dalvik.system.NativeStart.main(Native Method)`

the request for FEATURE_ACTION_BAR is just for API>11, but where is the actionbar added on lower apis or how to remove it?

it is a more complicated issue...

**Pull request closed without merging**